pkg/controller/sriovoperatorconfig/sriovoperatorconfig_controller.go

```diff
@@ -0,0 +1,329 @@
+package sriovoperatorconfig
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	sriovnetworkv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+	apply "github.com/openshift/sriov-network-operator/pkg/apply"
+	render "github.com/openshift/sriov-network-operator/pkg/render"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_sriovoperatorconfig")
+
+const (
+	DEFAULT_CONFIG_NAME	    = "default"
+	WEBHOOK_PATH                = "./bindata/manifests/webhook"
+	SERVICE_CA_CONFIGMAP        = "openshift-service-ca"
+	SRIOV_MUTATING_WEBHOOK_NAME = "network-resources-injector-config"
+)
+
+var Namespace = os.Getenv("NAMESPACE")
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new SriovOperatorConfig Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileSriovOperatorConfig{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("sriovoperatorconfig-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource SriovOperatorConfig
+	err = c.Watch(&source.Kind{Type: &sriovnetworkv1.SriovOperatorConfig{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner SriovOperatorConfig
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
```

Comment:
I think we shall watch the deployment instead of the pod. And also We may also need to watch the other resources (ServiceAccount, RBAC, ConfigMap, WebhookConfig). To prevent these resources from being changed unexpectedly.

Reply:
fixed